NLSY/nlsy_impute.R

```diff
@@ -0,0 +1,160 @@
+#Packages
+librarian::shelf(dplyr, mice, car, corrplot, caret)
+
+# Read data
+nlsydf = readRDS(paste0(here::here(), "/NLSY/NLSY-college-finance.rds"))
+
+test <- nlsydf %>% #Add additional variables that should be considered during multiple imputation
+    select(id = "PUBID_1997",
+           mom_age_birth = "CV_BIO_MOM_AGE_YOUTH_1997", # numeric
+           age_resp =  "CV_AGE_12/31/96_1997",
+           hispanic = "KEY_ETHNICITY_1997", # binary
+           race = "KEY_RACE_1997", # categorical
+           mom_education = "CV_HGC_RES_MOM_1997", # categorical - highest grade completed in 1997 by mom
+           dad_education = "CV_HGC_RES_DAD_1997", # categorical - highest grade completed in 1997 by dad
+           savings = "P5-130_1997", #binary - yes/no retirement savings
+           home = "P5-101_1997", # binary - homeownership or renting
+           both_parents = "YOUTH_BOTHBIO.01_1997",
+           par1_deceased = "YOUTH_NONR1DEAD.01_1997",
+           par2_deceased = "YOUTH_NONR2DEAD.01_1997",
+           pincome = "CV_INCOME_GROSS_YR_1997",
+           pnetworth = "CV_HH_NET_WORTH_P_1997",
+           # Also, it seems that this represents the wealth of the household in which a respondent lives. That means that for a respondent who lives with only one parent, the net-worth variable represents the wealth of that parent.
+           # In the case of an independent respondent, contains respondent's, rather than parents', wealth.
+    ) %>%
+    mutate(mom_age = mom_age_birth + age_resp)
+
+# Making dummy variables
+test$hisp <- ifelse(test$hispanic == "Yes", 1, 0)
+test$ret_sav <- ifelse(test$savings == "YES", 1, 0)
+test$hown <- ifelse(test$home == "OWNS OR IS BUYING; LAND CONTRACT", 1, 0)
+test$both_pars <- ifelse(test$both_parents == "Yes", 1, 0)
+test$par_dec <- ifelse(test$par1_deceased == "Yes" | test$par2_deceased == "Yes", 1, 0) # Either parent deceased
+test$mom_educ_hs <- ifelse(test$mom_education == "12TH GRADE" | test$mom_education == "1ST YEAR COLLEGE" | test$mom_education == "2ND YEAR COLLEGE" |  test$mom_education == "3RD YEAR COLLEGE" |  test$mom_education == "4TH YEAR COLLEGE" |  test$mom_education == "5TH YEAR COLLEGE" |  test$mom_education == "6TH YEAR COLLEGE" |  test$mom_education == "7TH YEAR COLLEGE" |  test$mom_education == "8TH YEAR COLLEGE", 1, 0)
+test$dad_educ_hs <- ifelse(test$dad_education == "12TH GRADE" | test$dad_education == "1ST YEAR COLLEGE" | test$dad_education == "2ND YEAR COLLEGE" |  test$dad_education == "3RD YEAR COLLEGE" |  test$mom_education == "4TH YEAR COLLEGE" |  test$mom_education == "5TH YEAR COLLEGE" |  test$mom_education == "6TH YEAR COLLEGE" |  test$mom_education == "7TH YEAR COLLEGE" |  test$mom_education == "8TH YEAR COLLEGE", 1, 0)
```

Comment:
A couple of comments on the education dummies:

1. It would be good to have more than two education categories. In addition to "high-school graduates", I would add "some college", "BS degree", "graduate degree".
2. We might want to experiment with different education groupings, which is why it would be good to implement this as a function in nlsy_lib.R. For example, nlsy_encode_educ4 for a function that encodes 4 categories. Having a function is also good because we will want to do this multiple times (for mother, father, and later for the respondent), and we don't want to copy-paste the code.
3. Rather than creating dummies, we can create factor variables, which R, in regressions, converts into dummies. 
4. Rather than comparing a variable to each of the values, a more efficient and less error-prone way to implement this type of encoding is by using the operator %in%.
 
The code that implements all of the suggestions above could look like this:
```
nlsy_encode_educ4 = function(var)
{
    return( case_when(
        {{var}} %in% c(
            "12TH GRADE"
        ) ~ "High-school graduate",
        {{var}} %in% c(
            "1ST YEAR COLLEGE",
            "2ND YEAR COLLEGE",
            "3RD YEAR COLLEGE"
        ) ~ "Some college",
        {{var}} %in% c(
            "4TH YEAR COLLEGE",
            "5TH YEAR COLLEGE"
        ) ~ "College degree",
        {{var}} %in% c(
            "6TH YEAR COLLEGE",
            "7TH YEAR COLLEGE",
            "8TH YEAR COLLEGE"
        ) ~ "Graduate degree"
    )
    )
}

test |>
    mutate(
        mom_educ_cat = factor(nlsy_encode_educ4(mom_education)),
        dad_educ_cat = factor(nlsy_encode_educ4(dad_education))
    )
 ```

Reply:
Seconded on all fronts here!

Reply:
Awesome, thank you for the explanation and the code sample. I have added it to the `nlsy_lib.R` script and called it in the imputation script to generate the factor versions of the education variables.